File: Taller de Haskell/Problema_5/problema_5.hs
-- Definimos la funcion 'obligacionFiscal'
-- Toma dos parámetros:
-- edad: un número entero (Int)
-- ingresos: un número decimal (Double)
-- Devuelve True si la persona tiene 18 años o más y gana mas de $5.500.000 COP
-- De lo contrario, devuelve False
obligacionFiscal :: Int -> Double -> Bool
obligacionFiscal edad ingresos = edad >= 18 && ingresos > 5500000
-- Función principal del programa
-- Solicita al usuario su edad e ingresos, y muestra si debe pagar impuestos
main :: IO ()
main = do
-- Solicita la edad del usuario
    putStrLn "Ingrese su edad:"
    edadStr <- getLine
    let edad = read edadStr :: Int
-- Solicita los ingresos mensuales del usuario
    putStrLn "Ingrese sus ingresos mensuales en COP:"
    ingresosStr <- getLine
    let ingresos = read ingresosStr :: Double
-- Evalúa si el usuario debe pagar impuestos usando la función obligacionFiscal
    print (obligacionFiscal edad ingresos)

File: Taller de Haskell/Problema_5/problema_5.py
# Solicita al usuario que ingrese su edad y convierte la entrada a un número entero
edad = int(input("Ingrese su edad: "))    
# Solicita al usuario que ingrese sus ingresos mensuales en pesos colombianos (COP)
# Convierte la entrada a un número decimal (float) para permitir valores con centavos
ingresos = float(input("Ingrese sus ingresos mensuales en COP: "))
# Verifica si el usuario cumple con los criterios para pagar impuestos:
# Tener 18 años o más
# Tener ingresos mensuales superiores a $5,500,000 COP
if edad >= 18 and ingresos > 5500000:
    # Si cumple ambos criterios, se indica que debe pagar impuestos
    print("Debe pagar impuestos.")
else:
    # Si no cumple alguno de los criterios, se indica que no está obligado a pagar impuestos
    print("No está obligado a pagar impuestos.")
